examples/Benchmark/generic/scenario_replication.h

```diff
@@ -0,0 +1,102 @@
+/*******************************************************************************
+ The MIT License (MIT)
+
+ Copyright (c) 2017 Grigory Nikolaenko <nikolaenko.grigory@gmail.com>
+
+ Permission is hereby granted, free of charge, to any person obtaining a copy
+ of this software and associated documentation files (the "Software"), to deal
+ in the Software without restriction, including without limitation the rights
+ to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ copies of the Software, and to permit persons to whom the Software is
+ furnished to do so, subject to the following conditions:
+
+ The above copyright notice and this permission notice shall be included in all
+ copies or substantial portions of the Software.
+
+ THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ SOFTWARE.
+ *******************************************************************************/
+
+#ifndef BENCHMARK_SCENARIO_REPLICATION_H
+#define BENCHMARK_SCENARIO_REPLICATION_H
+
+#include "benchmark.h"
+
+#include "../replication/generate_stream.h"
+#include "../../../Blocks/HTTP/api.h"
+#include "../../../Sherlock/replicator.h"
+
+#include "../../../Bricks/dflags/dflags.h"
+
+#ifndef CURRENT_MAKE_CHECK_MODE
+DEFINE_string(remote_url, "", "Remote url to subscribe to.");
+DEFINE_uint16(local_port, 8383, "Local port to spawn the source stream on.");
+DEFINE_string(db, "db.json", "Path to load the source stream data from.");
+DEFINE_bool(regenerate_db, true, "Regenerate the stream data to replicate.");
```

Comment:
The idea was to make sure there's no "read-write" flag.

in the present setup it's "--db", which _may or may not_ be overwritten if another flag is set. I may add the other flag forgetting of `--db` being set earlier in the command (stretch, I know, but still), and surprise myself later on realizing I've overwritten the DB file, possibly an important one. Practically speaking, I'd be reluctant to use `--db` of this tool to point to some prod database file, just because I'm too scared it can be overwritten.

And two flags eliminate such a risk altogether. That was my thinking :)